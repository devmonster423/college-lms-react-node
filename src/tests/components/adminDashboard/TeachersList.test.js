import React from 'react';
import { shallow } from 'enzyme';
import './../../__mocks__/localStorage';
import axios from './../../__mocks__/axios';
import TeacherList from './../../../components/adminDashboard/TeachersList';

describe('<TeachersList />', () => {
  test('should render the component properly w/ Loading', () => {
    const wrapper = shallow(<TeacherList />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should call the axios and with correct arguments', () => {
    expect(axios.called).toBeTruthy();
    expect(
      axios.calledWith({
        method: 'post',
        url: '/s/admin/giveallteacher',
        data: {
          token: localStorage.getItem('adminToken'),
        },
        headers: {
          'Content-Type': 'application/json',
        },
      })
    ).toBeTruthy();
  });

  // test('should render the components correctly with the data passed. ', (done) => {
  //   setTimeout(() => {
  //     const wrapper = shallow(<TeacherList />);
  //     expect(wrapper).toMatchSnapshot();
  //     done();
  //   }, 5000);
  // });
});
