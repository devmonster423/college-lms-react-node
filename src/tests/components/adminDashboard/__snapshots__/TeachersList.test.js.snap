// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeachersList /> should render the component properly w/ Loading 1`] = `
<styled.div
  h="450px"
  oflowy="scroll"
  p="10px"
>
  <p>
    Loading ...
  </p>
</styled.div>
`;
